debug/debug.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package os provides gNOI os operations.
+package debug
+
+import (
+	"context"
+
+	dbgpb "github.com/openconfig/gnoi/debug"
+	"github.com/openconfig/gnoigo/internal"
+)
+
+// DebugOperation represents the parameters of a Debug operation.
+// Debug will execute the set of commands provided in the request.
+// The command will be executed in the mode provided.
+// All command modes must support an exit code on completion of the
+// command. (e.g. Cli modes must exit after sending a command)
+// Errors:
+//
+//	InvalidArgument: for unspecified mode
+type DebugOperation struct {
+	req *dbgpb.DebugRequest
+}
+
+// Mode the commands will be executed in.
+//
+//	enum Mode {
```

Comment:
Putting the full enum in the comment creates a double-maintenance burden. I'd just let people look it up in the proto.